autopilot/agent_test.go

```diff
@@ -195,57 +222,95 @@ func TestAgentChannelOpenSignal(t *testing.T) {
 		Graph:       memGraph,
 		Constraints: constraints,
 	}
-	initialChans := []Channel{}
+
 	agent, err := New(testCfg, initialChans)
 	if err != nil {
 		t.Fatalf("unable to create agent: %v", err)
 	}
+	ctx.agent = agent
 
-	// To ensure the heuristic doesn't block on quitting the agent, we'll
-	// use the agent's quit chan to signal when it should also stop.
-	heuristic.quit = agent.quit
-
-	// With the autopilot agent and all its dependencies we'll star the
+	// With the autopilot agent and all its dependencies we'll start the
 	// primary controller goroutine.
 	if err := agent.Start(); err != nil {
 		t.Fatalf("unable to start agent: %v", err)
 	}
-	defer agent.Stop()
 
-	// We must defer the closing of quit after the defer agent.Stop(), to
-	// make sure ConnectToPeer won't block preventing the agent from
-	// exiting.
-	defer close(quit)
+	cleanup := func() {
+		// We must close quit before agent.Stop(), to make sure
+		// ChannelBudget won't block preventing the agent from exiting.
+		close(quit)
+		agent.Stop()
+	}
 
-	// We'll send an initial "no" response to advance the agent past its
-	// initial check.
+	return ctx, cleanup
+}
+
+// respondMoreChans consumes the moreChanArgs element and responds to the agent
+// with the given moreChansResp.
+func respondMoreChans(t *testing.T, testCtx *testContext, resp moreChansResp) {
```

Comment:
Nit: can use `t.Helper` here as well as in the rest of the assertion functions.

Reply:
Fixed: https://github.com/lightningnetwork/lnd/pull/2633/commits/8d148e78a7fcf865789724f573d97aa91ea1b537